### PR TITLE
Publish repository-hosted dependency binaries (build/packaging)

### PR DESCRIPTION
**Pull request by leynos**
## Summary
- Extend release and rolling-release workflows to build, archive, publish, and attach provenance/licence metadata for dependency binaries alongside existing Whitaker artefacts.
- Provide end-to-end support for repository-first installation of dependency binaries, with Cargo-based fallbacks preserved.
- Implement repository-hosted dependency binaries by introducing a committed manifest (installer/dependency-binaries.toml), a Rust domain model and packaging helper modules, and a packaging executable to publish and verify prebuilt binaries for cargo-dylint and dylint-link.
- Move beyond planning to implement the feature surface and its supporting tooling.

## Changes
- Documentation and ExecPlan: added docs/execplans/install-dependency-binaries.md describing the living plan and acceptance criteria.
- Manifest and domain model:
  - installer/dependency-binaries.toml (repository-owned manifest)
  - installer/src/dependency_binaries/mod.rs
  - installer/src/dependency_binaries/manifest.rs
  - installer/src/dependency_binaries/install.rs
- Packaging and artefact generation:
  - installer/src/dependency_packaging.rs
  - installer/src/bin/package_dependency_binary.rs (CLI to package binaries and provenance)
  - installer/src/dependency_packaging_tests.rs
  - installer/src/dependency_binaries/tests.rs (unit tests for manifest parsing)
- Installer wiring and orchestration:
  - installer/src/deps.rs (updated orchestration to prefer repository-hosted binaries first, with Cargo fallbacks)
  - installer/src/lib.rs (exposes new dependency_binaries and dependency_packaging modules)
  - installer/src/main.rs (uses updated install flow: install_dylint_tools_with_output)
- Behavioural and feature tests:
  - installer/tests/behaviour_dependency_binaries.rs
  - installer/tests/features/dependency_binaries.feature
- Release and packaging integration:
  - .github/workflows/release.yml (build and publish dependency binaries per target)
  - .github/workflows/rolling-release.yml (detect changes to dependency-binaries.toml and publish binaries on manifest changes)
- Roadmap and design docs updates:
  - docs/roadmap.md (adds 4.3.2 entry for dependency binaries)
  - docs/whitaker-dylint-suite-design.md (adds Dependency binary releases decisions)
- Other supporting docs:
  - installer/dependency-binaries.toml (manifest for cargo-dylint and dylint-link)
- Rationale
  - Centralizing required dependency-tool versions and provenance in a repository improves auditability, reproducibility, and the installer’s ability to resolve dependencies without relying solely on external Cargo tooling. The plan and associated tooling ensure repository-hosted binaries are elected first, with clear fallbacks and provenance data published alongside releases.

## Plan overview (stages A–I)
- Stage A: Add the repository-owned dependency-binary manifest (installer/dependency-binaries.toml).
- Stage B: Introduce a Rust domain model for dependency binaries (manifest, install logic, mod.rs).
- Stage C: Extend packaging for dependency binaries (packaging module and CLI, archive formats).
- Stage D: Extend rolling-release.yml to publish binaries on manifest changes.
- Stage E: Extend release.yml to publish binaries on tagged releases.
- Stage F: Teach installer (deps.rs) to prefer repository-hosted binaries first.
- Stage G: Add unit and behavioural tests (rstest-bdd coverage).
- Stage H: Update docs and roadmap to reflect the new feature.
- Stage I: Run full quality gates (fmt, lint, test).

> Note: This ExecPlan is living and will be updated as design decisions solidify and implementation proceeds.

## Acceptance criteria
- A repository-owned manifest (installer/dependency-binaries.toml) is committed on main.
- A Rust domain model for dependency binaries is introduced and exercised by tests.
- Packaging tooling exists to build, archive, and publish dependency binaries alongside licences/provenance.
- Rolling-release and tagged-release workflows publish the dependency binaries and provenance artefact.
- The installer prefers repository-hosted binaries before cargo-based installation, with fallbacks preserved.
- Unit and behavioural tests cover success, fallback, and failure scenarios.
- Documentation and roadmap entries reflect the feature and its status.

## Testing plan
- The ExecPlan defines tests for manifest parsing, archive naming, provenance/licence generation, and installer orchestration.
- Behavioural tests with rstest-bdd validate missing assets, successful repository installs, and fallback paths.
- Full quality gates (fmt, lint, test) will be run as part of Stage I of implementation.

## Impact and next steps
- No partial changes; this PR implements the feature surface and supporting tooling. Subsequent PRs should refine security, edge cases, and test coverage across CI.

## Related files
- docs/execplans/install-dependency-binaries.md (new)
- installer/dependency-binaries.toml (new)
- installer/src/dependency_binaries/mod.rs (new)
- installer/src/dependency_binaries/manifest.rs (new)
- installer/src/dependency_binaries/install.rs (new)
- installer/src/dependency_packaging.rs (new)
- installer/src/bin/package_dependency_binary.rs (new)
- installer/src/dependency_packaging_tests.rs (new)
- installer/tests/behaviour_dependency_binaries.rs (new)
- installer/tests/features/dependency_binaries.feature (new)
- .github/workflows/release.yml (updated)
- .github/workflows/rolling-release.yml (updated)
- docs/roadmap.md (updated)
- docs/whitaker-dylint-suite-design.md (updated)
- installer/dependency-binaries.toml (new)

◳ Generated by [DevBoxer](https://www.devboxer.com) ◰

📎 **Task**: https://www.devboxer.com/task/ecea5f11-c182-4674-bae9-6c553a267773

### CODE REVIEW

File: installer/src/test_utils.rs [hunk 1]
```diff
@@ -262,3 +285,164 @@ impl CommandExecutor for StubExecutor {
...
+
+    /// Creates an expected call for verifying dylint-link installation.
+    pub fn dylint_link_check() -> ExpectedCall {
+        ExpectedCall {
+            cmd: "dylint-link",
+            args: vec!["--version"],
+            result: Ok(success_output()),
+        }
+    }
+}
```
[the remark this answers is not in the document]
Comment (leynos) [reply installer/src/test_utils.rs]: @coderabbitai Has this now been resolved in the latest commit?

Use codegraph analysis to determine your answer.

If this comment is now resolved, please mark it as such using the API. Otherwise, please provide an AI agent prompt for the remaining work to be done to address this comment.

File: installer/src/dependency_binaries/install/tests.rs [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+                    binary: expected_member_path.to_owned(),
+                });
+            }
+            let binary_name = Path::new(expected_member_path)
+                .file_name()
+                .expect("member path should include a filename");
+            let path = destination_dir.join(binary_name);
+            fs::write(&path, b"fake binary")?;
+            Ok(path)
+        });
```
[the remark this answers is not in the document]
Comment (leynos) [reply installer/src/dependency_binaries/install/tests.rs]: @coderabbitai Has this now been resolved in the latest commit?

Use codegraph analysis to determine your answer.

If this comment is now resolved, please mark it as such using the API. Otherwise, please provide an AI agent prompt for the remaining work to be done to address this comment.

File: installer/scripts/dependency_binaries_manifest.py [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+
+    Raises
+    ------
+    FileNotFoundError, OSError
+        If the manifest file cannot be opened.
+    tomllib.TOMLDecodeError
+        If the manifest contains invalid TOML syntax.
+    KeyError
+        If required fields are absent.
+    """
```
[the remark this answers is not in the document]
Comment (leynos) [reply installer/scripts/dependency_binaries_manifest.py]: @coderabbitai Has this now been resolved in the latest commit?

Use codegraph analysis to determine your answer.

If this comment is now resolved, please mark it as such using the API. Otherwise, please provide an AI agent prompt for the remaining work to be done to address this comment.
